src/Rules/Documentation.hs

```diff
@@ -107,10 +109,14 @@ buildLibraryDocumentation :: Rules ()
 buildLibraryDocumentation = do
     root <- buildRootRules
     root -/- htmlRoot -/- "libraries/index.html" %> \file -> do
+        need [ root -/- "stage1/lib/llvm-targets" ]
```

Comment:
Why do we *need* the `llvm-targets` file here?

Reply:
I don't know, but the docs rule used to get stuck on that before I added this line. I have to say I'm not entirely sure when this file is needed and by what.

Reply:
I can try leaving it out and seeing what goes wrong, at least so that I can leave a comment explaining why this is needed?

Reply:
Yep, that would be good. Esp, as the `llvm-targets` is a copy target (I believe) I don't really see what it would pull in that would be required here. I guess it might transitively require the stage0 compiler. Then again I'm not sure why that one is needed.

Reply:
Interesting. That suggests that haddock actually uses the `compiler` library... maybe we *need* `ghc` for haddock? Or does that sound wrong?

Reply:
Maybe... I think it uses GHC's argument parsing to parse the GHC arguments given to haddock. And maybe more.

Reply:
Would that justify it? I have no idea where that llvm-targets file is used for now.

Reply:
Guess, it's good for now. The llvm-targets is read during dynflag construction iirc.

Reply:
Oh then it would make sense, given that Haddock uses GHC's code for parsing haddock CLI arguments that are destined to GHC.